File: forth/009.fs
#! /usr/bin/gforth

: pe009
1000 1 do
  1000 i do
    1000 j do
     i j k + + 1000 =
     if
       k k * j j * + i i * =
       if 
         i j k * * . cr
         bye
       then
     then
    loop
  loop
loop
;

pe009
bye

File: perl6/009.pl
#!/usr/bin/perl -w
use strict;

my $stop = 1000;
loop (my $a = 1; $a < $stop; $a++) {
  loop (my $b = $a + 1; $b < $stop; $b++) {
    loop (my $c = $b + 1; $c < $stop; $c++) {
      if ((($a + $b +$c) == $stop) and (($a * $a) + ($b * $b) == ($c * $c))) {
        say $a*$b*$c;
        exit(0);
      }
    }
  }
}
exit(1);
